### ISSUE TITLE
Neos Setup endless switches between login screen and "Database Error" screen

### ISSUE THREAD

**Issue by InvisibleKind** [neos/neos-development-collection#1691]
### Description

The issue is almost same as closed #1663 
On first setup right after login screen I get a "Database Error" message abd after clicking on "Go to setup" I land back to login screen, which again leads to same error.

### Steps to Reproduce

1. Fresh Neos 3.2 installation (also tried with latest 3.1)
2. Go to domain.tld/setup and wait for initial caches warmup
3. Login and see a "Database Error" message
4. Click on "Go to Setup"
5. Initializing setup screen followed by login is shown.
6. Login leads again to "Database error"

#### Expected behavior

Setup configuration screens are expected, incuding DB configuration.

#### Actual behavior

Erorr message, followed by login.

### Affected Versions

Neos: 3.2.0, 3.1.5

### Additional info

Environment: Vagrant
Vagrant Box: Laravel Homestead 3.0.0

` tail -f Data/Logs/System_Development.log` shows the lats entry:
`17-09-03 08:26:33 8109       CRITICAL  DBAL                 Exception in line 145 of /home/vagrant/myproj/Packages/Libraries/doctrine/dbal/lib/Doctrine/DBAL/DBALException.php: An exception occured in driver: SQLSTATE[HY000] [1045] Access denied for user ''@'localhost' (using password: NO) - See also: 20170903082632b48c83.txt
    previousException => Exception #1045 in line 41 of /home/vagrant/myproj/Packages/Libraries/doctrine/dbal/lib/Doctrine/DBAL/Driver/PDOMySql/Driver.php: SQLSTATE[HY000] [1045] Access denied for user ''@'localhost' (using password: NO)`

**Comment by iseries** [comment]
I wonder why there is no mysql user given (in the exception log).  Im not an expert but i think you have to check your settings.yaml and add an existing mysql user. 

Hope this helps

**Comment by InvisibleKind** [comment]
@iseries the problem is, that it is a fresh installation and no settings have been set yet - I expect them to be requested on next screens.

Meanwhile I've found the reason of error. Environemnt variable `FLOW_REWRITEURLS` need to be set to `1`.
Since Laravel Homeastead uses Nginx instead of Apache, this variabe was not automatically enabled via .htaccess.

I think, this setting worth mentioning for non-Apache setups in [Installation Guide](http://neos.readthedocs.io/en/stable/GettingStarted/Installation.html).

**Comment by nrueckmann** [comment]
Still happens (at least for me) running in Nginx Unit Docker container

```
Exception in line 182 of /neos/Packages/Libraries/doctrine/dbal/lib/Doctrine/DBAL/DBALException.php: An exception occurred in driver: SQLSTATE[HY000] [2002] No such file or directory

28 Doctrine\DBAL\Driver\AbstractMySQLDriver::convertException("An exception occurred in driver: SQLSTATE[HY000] [2002] No such file or directory", Doctrine\DBAL\Driver\PDO\Exception)
27 Doctrine\DBAL\DBALException::wrapException(Doctrine\DBAL\Driver\PDO\MySQL\Driver, Doctrine\DBAL\Driver\PDO\Exception, "An exception occurred in driver: SQLSTATE[HY000] [2002] No such file or directory")
26 Doctrine\DBAL\DBALException::driverException(Doctrine\DBAL\Driver\PDO\MySQL\Driver, Doctrine\DBAL\Driver\PDO\Exception)
25 Doctrine\DBAL\Driver\PDOMySql\Driver::connect(array|8|, NULL, NULL, array|0|)
24 Doctrine\DBAL\Connection::connect()
23 Doctrine\DBAL\Connection::getWrappedConnection()
22 Doctrine\DBAL\Connection::executeQuery("SELECT 1 FROM neos_flow_security_account t0 WHERE t0.persistence_object_identifier = ?", array|1|, array|1|)
21 Doctrine\DBAL\Connection::fetchOne("SELECT 1 FROM neos_flow_security_account t0 WHERE t0.persistence_object_identifier = ?", array|1|, array|1|)
20 Doctrine\ORM\Persisters\Entity\BasicEntityPersister::exists(Neos\Flow\Security\Account)
19 Doctrine\ORM\UnitOfWork::getEntityState(Neos\Flow\Security\Account)
18 Neos\Flow\Persistence\Doctrine\PersistenceManager_Original::isNewObject(Neos\Flow\Security\Account)
17 Neos\Flow\Security\Authentication\Token\PasswordToken::Flow_serializeRelatedEntities(array|1|, array|7|)
16 Neos\Flow\Security\Authentication\Token\PasswordToken::__sleep()
15 serialize(array|1|)
14 Neos\Cache\Frontend\VariableFrontend::set("7681d4d4-4f0b-44d6-89ea-908ca9e4a280d1993ff050a2c957f03196d5a89730c9", array|1|, array|1|, 0)
13 Neos\Flow\Session\Session_Original::putData("Neos_Flow_Object_ObjectManager", array|1|)
12 Neos\Flow\Session\Session_Original::shutdownObject()
11 Neos\Flow\ObjectManagement\ObjectManager::callShutdownMethods(SplObjectStorage)
10 Neos\Flow\ObjectManagement\ObjectManager::Neos\Flow\ObjectManagement\{closure}()
9 Closure::__invoke()
8 Neos\Flow\Security\Context_Original::withoutAuthorizationChecks(Closure)
7 Neos\Flow\ObjectManagement\ObjectManager::shutdown("Runtime", "Neos\Flow\Core\Bootstrap::bootstrapShuttingDown")
6 call_user_func_array(array|2|, array|2|)
5 Neos\Flow\SignalSlot\Dispatcher::dispatch("Neos\Flow\Core\Bootstrap", "bootstrapShuttingDown", array|1|)
4 Neos\Flow\Core\Bootstrap::emitBootstrapShuttingDown("Runtime")
3 Neos\Flow\Core\Bootstrap::shutdown("Runtime")
2 Neos\Setup\Core\RequestHandler::handleRequest()
1 Neos\Flow\Core\Bootstrap::run()

Previous exception: Exception #2002 in line 44 of /neos/Packages/Libraries/doctrine/dbal/lib/Doctrine/DBAL/Driver/PDOConnection.php: SQLSTATE[HY000] [2002] No such file or directory

27 Doctrine\DBAL\Driver\PDO\Exception::new(PDOException)
26 Doctrine\DBAL\Driver\PDOConnection::__construct("mysql:charset=utf8mb4;", NULL, NULL, array|0|)
25 Doctrine\DBAL\Driver\PDOMySql\Driver::connect(array|8|, NULL, NULL, array|0|)
24 Doctrine\DBAL\Connection::connect()
23 Doctrine\DBAL\Connection::getWrappedConnection()
22 Doctrine\DBAL\Connection::executeQuery("SELECT 1 FROM neos_flow_security_account t0 WHERE t0.persistence_object_identifier = ?", array|1|, array|1|)
21 Doctrine\DBAL\Connection::fetchOne("SELECT 1 FROM neos_flow_security_account t0 WHERE t0.persistence_object_identifier = ?", array|1|, array|1|)
20 Doctrine\ORM\Persisters\Entity\BasicEntityPersister::exists(Neos\Flow\Security\Account)
19 Doctrine\ORM\UnitOfWork::getEntityState(Neos\Flow\Security\Account)
18 Neos\Flow\Persistence\Doctrine\PersistenceManager_Original::isNewObject(Neos\Flow\Security\Account)
17 Neos\Flow\Security\Authentication\Token\PasswordToken::Flow_serializeRelatedEntities(array|1|, array|7|)
16 Neos\Flow\Security\Authentication\Token\PasswordToken::__sleep()
15 serialize(array|1|)
14 Neos\Cache\Frontend\VariableFrontend::set("7681d4d4-4f0b-44d6-89ea-908ca9e4a280d1993ff050a2c957f03196d5a89730c9", array|1|, array|1|, 0)
13 Neos\Flow\Session\Session_Original::putData("Neos_Flow_Object_ObjectManager", array|1|)
12 Neos\Flow\Session\Session_Original::shutdownObject()
11 Neos\Flow\ObjectManagement\ObjectManager::callShutdownMethods(SplObjectStorage)
10 Neos\Flow\ObjectManagement\ObjectManager::Neos\Flow\ObjectManagement\{closure}()
9 Closure::__invoke()
8 Neos\Flow\Security\Context_Original::withoutAuthorizationChecks(Closure)
7 Neos\Flow\ObjectManagement\ObjectManager::shutdown("Runtime", "Neos\Flow\Core\Bootstrap::bootstrapShuttingDown")
6 call_user_func_array(array|2|, array|2|)
5 Neos\Flow\SignalSlot\Dispatcher::dispatch("Neos\Flow\Core\Bootstrap", "bootstrapShuttingDown", array|1|)
4 Neos\Flow\Core\Bootstrap::emitBootstrapShuttingDown("Runtime")
3 Neos\Flow\Core\Bootstrap::shutdown("Runtime")
2 Neos\Setup\Core\RequestHandler::handleRequest()
1 Neos\Flow\Core\Bootstrap::run()

Previous exception: Exception #2002 in line 40 of /neos/Packages/Libraries/doctrine/dbal/lib/Doctrine/DBAL/Driver/PDOConnection.php: SQLSTATE[HY000] [2002] No such file or directory

27 PDO::__construct("mysql:charset=utf8mb4;", "", "", array|0|)
26 Doctrine\DBAL\Driver\PDOConnection::__construct("mysql:charset=utf8mb4;", NULL, NULL, array|0|)
25 Doctrine\DBAL\Driver\PDOMySql\Driver::connect(array|8|, NULL, NULL, array|0|)
24 Doctrine\DBAL\Connection::connect()
23 Doctrine\DBAL\Connection::getWrappedConnection()
22 Doctrine\DBAL\Connection::executeQuery("SELECT 1 FROM neos_flow_security_account t0 WHERE t0.persistence_object_identifier = ?", array|1|, array|1|)
21 Doctrine\DBAL\Connection::fetchOne("SELECT 1 FROM neos_flow_security_account t0 WHERE t0.persistence_object_identifier = ?", array|1|, array|1|)
20 Doctrine\ORM\Persisters\Entity\BasicEntityPersister::exists(Neos\Flow\Security\Account)
19 Doctrine\ORM\UnitOfWork::getEntityState(Neos\Flow\Security\Account)
18 Neos\Flow\Persistence\Doctrine\PersistenceManager_Original::isNewObject(Neos\Flow\Security\Account)
17 Neos\Flow\Security\Authentication\Token\PasswordToken::Flow_serializeRelatedEntities(array|1|, array|7|)
16 Neos\Flow\Security\Authentication\Token\PasswordToken::__sleep()
15 serialize(array|1|)
14 Neos\Cache\Frontend\VariableFrontend::set("7681d4d4-4f0b-44d6-89ea-908ca9e4a280d1993ff050a2c957f03196d5a89730c9", array|1|, array|1|, 0)
13 Neos\Flow\Session\Session_Original::putData("Neos_Flow_Object_ObjectManager", array|1|)
12 Neos\Flow\Session\Session_Original::shutdownObject()
11 Neos\Flow\ObjectManagement\ObjectManager::callShutdownMethods(SplObjectStorage)
10 Neos\Flow\ObjectManagement\ObjectManager::Neos\Flow\ObjectManagement\{closure}()
9 Closure::__invoke()
8 Neos\Flow\Security\Context_Original::withoutAuthorizationChecks(Closure)
7 Neos\Flow\ObjectManagement\ObjectManager::shutdown("Runtime", "Neos\Flow\Core\Bootstrap::bootstrapShuttingDown")
6 call_user_func_array(array|2|, array|2|)
5 Neos\Flow\SignalSlot\Dispatcher::dispatch("Neos\Flow\Core\Bootstrap", "bootstrapShuttingDown", array|1|)
4 Neos\Flow\Core\Bootstrap::emitBootstrapShuttingDown("Runtime")
3 Neos\Flow\Core\Bootstrap::shutdown("Runtime")
2 Neos\Setup\Core\RequestHandler::handleRequest()
1 Neos\Flow\Core\Bootstrap::run()


HTTP REQUEST:
target: /setup/login/authenticate
Host: localhost:8080
Connection: keep-alive
Cache-Control: max-age=0
Sec-Ch-Ua: "Google Chrome";v="111", "Not(A:Brand";v="8", "Chromium";v="111"
Sec-Ch-Ua-Mobile: ?0
Sec-Ch-Ua-Platform: "macOS"
Upgrade-Insecure-Requests: 1
Origin: http://localhost:8080
User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/111.0.0.0 Safari/537.36
Accept: text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.7
Sec-Fetch-Site: same-origin
Sec-Fetch-Mode: navigate
Sec-Fetch-User: ?1
Sec-Fetch-Dest: document
Referer: http://localhost:8080/setup/login?step=0
Accept-Encoding: gzip, deflate, br
Accept-Language: de,en-US;q=0.9,en;q=0.8
Cookie: Phpstorm-34b276c6=e82d0d37-fdad-4add-a8a0-f67ab947f661; Neos_Session=lVr5tbx0Jy25OnRREoARr7s0TuIwf7vB
Content-Length: 811
Content-Type: application/x-www-form-urlencoded

PHP PROCESS:
Inode: 
PID: 93
UID: 101
GID: 101
User: 

    [array] => 
        previousException:
        [string] => Exception #2002 in line 44 of /neos/Packages/Libraries/doctrine/dbal/lib/Doctrine/DBAL/Driver/PDOConnection.php: SQLSTATE[HY000] [2002] No such file or directory
```

**Comment by nrueckmann** [comment]
Same happens on Apache

**Comment by mhsdesign** [comment]
Hmm i strongly recommend to not use the visual gui setup.
Since some time we introduced cli based setup command type `./flow welcome` to see if they are available and continue the suggested ones to setup the db and other things ;)

**Comment by olimination** [comment]
I can reproduce the same issue going through the latest version of the Docker setup guide: https://docs.neos.io/guide/installation-development-setup/docker-and-docker-compose-setup

@mhsdesign : Maybe it would make sense to update section "5. Go through the setup tool" with some hint by using the new cli-setup approach? What do you think? And is the "GUI Setup tool" still a valid/supported approach or is it more like "deprecated"? If yes then such an info could also make sense.

**Comment by mhsdesign** [comment]
Yes I’m very much in favour of this idea. @olimination

https://neos-project.slack.com/archives/C04Q0TC15/p1680685799429459 But there are also voices against this …

**Comment by tobiasgruber** [comment]
There just was a discussion at the Sprint in Berlin where a [new approach to the Neos Setup](https://neos-project.slack.com/archives/C0L41E268/p1682935777670669) was discussed and decided to be implemented.

TLDR; a visual entry point is expected and useful, there will be one setup method via the cli, the visual setup is enhanced to a healthcheck
